### PR TITLE
Document manifest files

### PR DESCRIPTION
**Pull request by pepicrft**
As part of the documentation revamp, I forgot to include a page explaining the different supported manifest files. This PR includes that page and also provides some guidelines for deciding between "mono or multi-project" setups, which is a recurrent question asked by users.

### CODE REVIEW

File: docs/docs/guide/project/manifests.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    name: "App",
+    targets: [
+        // ....
+    ]
+)
+```
+
+
+> [!WARNING] ROOT VARIABLES
+> The only variable that should be at the root of the manifest is `let project = Project(...)`. If you need to reuse code across various parts of the manifest, you can use Swift functions.
```
Comment (danieleformichelli) [inline docs/docs/guide/project/manifests.md:31]: Why this? We have a lot of top-level variable and never had issues with it AFAIK

File: docs/docs/guide/project/manifests.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+let workspace = Workspace(
+    name: "App-Workspace",
+    projects: [
+        "./App", // Path to directory containing the Project.swift file
+    ]
+)
+```
+
+> [!NOTE]
+> Tuist will resolve the dependency graph and include the projects of the dependencies in the workspace. You don't need to include them manually.
```
Comment (danieleformichelli) [inline docs/docs/guide/project/manifests.md:50]: I think we should explain what's the point of defining them here then

**Review by danieleformichelli — APPROVED** [review]

File: docs/docs/guide/project/manifests.md (hunk 1 above)
Comment (pepicrft) [reply docs/docs/guide/project/manifests.md:31]: I'm not sure. I'll have to investigate the next time we get a report.

File: docs/docs/guide/project/manifests.md (hunk 2 above)
Comment (pepicrft) [reply docs/docs/guide/project/manifests.md]: Good idea. I added an additional sentence clarifying that.

**Pull request merged**